geomstats/geometry/special_linear.py

```diff
@@ -0,0 +1,211 @@
+    def __init__(self, n):
+        super(SpecialLinearLieAlgebra, self).__init__(
+            dim=n**2 - 1,
+            n=n,
+        )
+        # ???: not sure it is in the right place
+        self.ambient_space = GeneralLinear(n)
+
+        # create basis
+        rows, cols = gs.triu_indices(self.n, k=1)
+        diag_rows, diag_cols = gs.diag_indices(self.n)
+
+        dims = gs.arange(0, self.dim)
+        n_non_diag_half = (self.dim - n + 1) // 2
+
+        self.basis = gs.zeros((self.dim, n, n))
+
+        self.basis[dims[:n_non_diag_half], rows, cols] = 1.
+        self.basis[dims[n_non_diag_half:-n + 1], cols, rows] = 1.
+
+        self.basis[dims[-n + 1:], 0, 0] = -1
+        self.basis[dims[-n + 1:], diag_rows[1:], diag_cols[1:]] = 1
+
+    def basis_representation(self, matrix_representation):
+        """Compute the coefficients of matrices in the given basis.
+
+        Assume the basis is the one described in this answer on StackOverflow:
+        https://math.stackexchange.com/a/1949395/920836
+
+        Parameters
+        ----------
+        matrix_representation : array-like, shape=[..., n, n]
+            Matrix.
+
+        Returns
+        -------
+        basis_representation : array-like, shape=[..., dim]
+            Coefficients in the basis.
+        """
+        rows, cols = gs.triu_indices(self.n, k=1)
+        diag_rows, diag_cols = gs.diag_indices(self.n)
+
+        vec = gs.concatenate(
+            [
+                matrix_representation[..., rows, cols],
+                matrix_representation[..., cols, rows],
+                matrix_representation[..., diag_rows[1:], diag_cols[1:]],
+            ], axis=-1
+        )
+
+        return vec
+
+    def belongs(self, point, atol=gs.atol):
+        """Evaluate if the point belongs to the Lie algebra.
+
+        This method checks the shape of the input point and its trace.
+
+        Parameters
+        ----------
+        point : array-like, shape=[.., n, n]
+            Point to test.
+        atol : float
+            Tolerance threshold for zero values.
+
+        Returns
+        -------
+        belongs : array-like, shape=[...,]
+            Boolean evaluating if point belongs to the space.
+        """
+        has_right_shape = self.ambient_space.belongs(point)
+        if gs.all(has_right_shape):
+            return gs.isclose(gs.trace(point, axis1=-2, axis2=-1),
+                              0.0, atol=atol)
+
+        return has_right_shape
+
+    def projection(self, point):
+        """Project a point to the Lie algebra.
+
+        This can be done by removing the trace in the first entry of the matrix.
+
+        Parameters
+        ----------
+        point: array-like, shape=[..., n, n]
+            Point.
+
+        Returns
+        -------
+        point: array-like, shape=[..., n, n]
+            Projected point.
+        """
+        projected_point = gs.copy(point)
+        projected_point[..., 0, 0] = projected_point[..., 0, 0] - gs.trace(
```

Comment:
Assignment -> you can use the first element of the basis: `p -= trace * e0`

Reply:
this solves the issue (after `Matrices.basis` implementation `gs.array_from_sparse` can be replaced):

```
trace = gs.trace(point, axis1=-2, axis2=-1)
# TODO: use ambient_matrix basis instead
sps_matrix = gs.array_from_sparse([(0, 0)], [1.0], (self.n, self.n))

return gs.copy(point) - gs.einsum('...,...ij->...ij', trace, sps_matrix)
```